diff --git a/certs/extract-cert.c b/certs/extract-cert.c
index 5c3c881f1..da834c81c 100644
--- a/certs/extract-cert.c
+++ b/certs/extract-cert.c
@@ -149,8 +149,10 @@ int main(int argc, char **argv)
 			drain_openssl_errors();
 		else
 			ERR(1, "ENGINE_init");
+#ifdef USE_PKCS11_ENGINE
 		if (key_pass)
 			ERR(!ENGINE_ctrl_cmd_string(e, "PIN", key_pass, 0), "Set PKCS#11 PIN");
+#endif
 		ENGINE_ctrl_cmd(e, "LOAD_CERT_CTRL", 0, &parms, NULL, 1);
 		ERR(!parms.cert, "Get X.509 from PKCS#11");
 		write_cert(parms.cert);

diff --git a/fs/incfs/data_mgmt.c b/fs/incfs/data_mgmt.c
index 09959fac6..6368843fe 100644
--- a/fs/incfs/data_mgmt.c
+++ b/fs/incfs/data_mgmt.c
@@ -472,8 +472,11 @@ static ssize_t decompress(struct mount_info *mi,
 
 	switch (alg) {
 	case INCFS_BLOCK_COMPRESSED_LZ4:
-		result = LZ4_decompress_safe(src.data, dst.data, src.len,
-					     dst.len);
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+	result = LZ4_arm64_decompress_safe(src.data, dst.data, src.len, dst.len, false);
+#else
+	result = LZ4_decompress_safe(src.data, dst.data, src.len, dst.len);
+#endif
 		if (result < 0)
 			return -EBADMSG;
 		return result;

